src/apps/deskflow-gui/MainWindow.cpp

```diff
@@ -1026,3 +1058,19 @@ void MainWindow::showAndActivate()
   m_actionRestore->setVisible(false);
   m_actionMinimize->setVisible(true);
 }
+
+QString MainWindow::getTlsPath()
+{
+  CoreTool coreTool;
+  return QStringLiteral("%1/%2").arg(coreTool.getProfileDir(), kSslDir);
+}
+
+bool MainWindow::regenerateLocalFingerprints()
+{
+  TlsCertificate tls;
+  if (!tls.generateFingerprint(m_appConfig.tlsCertPath())) {
+    QMessageBox::critical(this, tr("TLS Fingerprint Error"), tr("Failed to calculate  keys"));
```

Comment:
Thought we were changing this to use `qCritical`?
```suggestion
    qCritical() << "failed to calculate new keys";
```

Reply:
Id rather show a dialog to the user.

Reply:
FTR, discussed on Matrix, `qCritical()` does both but we will fix later so we can land now.